flake.nix

```diff
@@ -11,66 +11,115 @@
       let
         pkgs = nixpkgs.legacyPackages.${system};
 
-        # Function to build the webapp with configurable API endpoint
-        buildWebapp = { apiEndpoint ? "https://api.moneybadger.sirion.io" }:
-          pkgs.stdenv.mkDerivation {
-            pname = "moneybadger-qr-scanner";
-            version = "0.1.0";
+        # Build the webapp
+        webapp = pkgs.stdenv.mkDerivation {
+          pname = "moneybadger-qr-scanner";
+          version = "0.1.0";
 
-            src = ./.;
+          src = ./.;
 
-            buildInputs = [ ];
+          buildPhase = ''
+            mkdir -p $out
+            cp favicon.ico $out/
+            cp index.html $out/
+          '';
 
-            buildPhase = ''
-              # Create output directory
-              mkdir -p $out
+          installPhase = "true";
 
-              # Copy favicon
-              cp favicon.ico $out/
+          meta = with pkgs.lib; {
+            description = "Moneybadger QR Scanner - Lightning payment scanner webapp";
+            license = licenses.mit;
+            platforms = platforms.all;
+          };
+        };
 
-              # Process index.html to replace API endpoint
-              substitute index.html $out/index.html \
-                --replace-fail "https://api.moneybadger.sirion.io" "${apiEndpoint}"
-            '';
+        # Nginx configuration for Docker
+        nginxConf = pkgs.writeText "nginx.conf" ''
+          user nobody nobody;
+          worker_processes 1;
+          error_log /dev/stderr;
+          pid /tmp/nginx.pid;
 
-            installPhase = "true"; # Already installed in buildPhase
+          events {
+            worker_connections 1024;
+          }
+
+          http {
+            include ${pkgs.nginx}/conf/mime.types;
+            default_type application/octet-stream;
+            access_log /dev/stdout;
+            sendfile on;
+
+            server {
+              listen 80;
+              root /var/www/html;
+              index index.html;
+
+              location / {
+                try_files $uri $uri/ /index.html;
+              }
+
+              location /api/ {
+                proxy_pass https://api.cryptoqr.co.za/;
+                proxy_http_version 1.1;
+                proxy_set_header Host api.cryptoqr.co.za;
+                proxy_set_header X-Real-IP $remote_addr;
+                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+                proxy_set_header X-Forwarded-Proto $scheme;
+                proxy_ssl_server_name on;
+                proxy_connect_timeout 30s;
+                proxy_read_timeout 60s;
+                proxy_send_timeout 60s;
+              }
```

Comment:
Nice, you were able to simplify the config a lot because it's not the same host. Great idea! Makes everything simpler. Totally evaded me while trying to solve one problem at a time 😆